internal/synclet/synclet.go

```diff
@@ -19,8 +23,70 @@ func NewSynclet(dcli build.DockerClient) *Synclet {
 	return &Synclet{dcli: dcli}
 }
 
-func (s Synclet) GetContainerIdForPod(podId string) (string, error) {
-	return "", errors.New("GetContainerIdForPod not implemented")
+const pauseCmd = "/pause"
+
+func (s Synclet) GetContainerIdForPod(ctx context.Context, podId string) (string, error) {
+	a := filters.NewArgs()
+	a.Add("name", podId)
+	listOpts := types.ContainerListOptions{Filters: a}
+	containers, err := s.dcli.ContainerList(ctx, listOpts)
+	if err != nil {
+		return "", fmt.Errorf("getting containers: %v", err)
+	}
+
+	if len(containers) == 0 {
+		return "", fmt.Errorf("no containers found with name %s", podId)
+	}
+
+	// We expect there to be one real match and one spurious match -- a container running
+	// "/pause" (see: https://www.ianlewis.org/en/almighty-pause-container); filter it out
+	if len(containers) > 2 {
+		var ids []string
+		for _, c := range containers {
+			ids = append(ids, c.ID[:10])
+		}
+		return "", fmt.Errorf("too many matching containers (%v)", ids)
+	}
+
+	for _, c := range containers {
+		// TODO(maia): more robust check here (what if user is running a container with "/pause" command?!)
+		if c.Command != pauseCmd {
+			return c.ID, nil
+		}
+	}
+
+	// What?? No actual matches??!
+	return "", fmt.Errorf("no real containers -- all were '/pause' containers")
+}
+
+func (s Synclet) writeFiles(ctx context.Context, containerId string, tarArchive []byte) error {
+	if tarArchive == nil {
+		return nil
+	}
+
+	err := s.dcli.CopyToContainerRoot(ctx, containerId, bytes.NewBuffer(tarArchive))
```

Comment:
why not just `return s.dcli.CopyTo...`?

Reply:
Mostly to minimize edits/diffs when adding more calls or reordering calls (in that one wouldn't have to change `return s.dcli.CopyTo` to `err = s.dcli`...`if err != nil`), for the same reason some languages allow trailing commas after the last item in a list, but I don't know what's idiomatic go and don't really feel strongly either way.

Reply:
eh. fair. doesn't really matter i guess.